src/main/scala/chisel3/util/circt/SizeOf.scala

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import scala.util.hashing.MurmurHash3
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule}
+
+import circt.Intrinsic
+
+// We have to unique designedName per type, be we can't due to name conflicts
+// on bundles.  Thus we use a globally unique id.
+private object SizeOfGlobalIDGen {
+  private var id: Int = 0
+  def getID() = {
+    this.synchronized {
```

Comment:
Nice use of synchronized.

Reply:
@jackkoenig  This is another example, where a Builder context cache would be useful. `synchronized` works here, but only because gaps in the IDs are acceptable.

Reply:
Agreed, can this instead use the same pattern as `Instantiate` which has [a cache that is part of the Builder](https://github.com/chipsalliance/chisel3/blob/df203ea8a7331ead3ed7a6e3ef2d0a5a8bb1a9f5/core/src/main/scala/chisel3/internal/Builder.scala#L553)? See https://github.com/chipsalliance/chisel3/pull/2717 for an example of what needs to be added.

Then at least we will have all examples that need the pattern grouped up so that we'll remember to fix them when we add the Builder context cache.

Reply:
is synchronized deterministic?

Reply:
> is synchronized deterministic?

IDs may be non-deterministic if you are elaborating more than one circuit in a single process

Reply:
hm, i do not like that. That is something we do, e.g. in testing code. Can we do what Jack suggested?